Remove calculate_status framework function and unittest test case

mcarmona99 opened the pull request:
|Related issue|
|---|
| #8802 |

Hi team,

This PR closes #8802,

In this pull request, I have removed the unused calculate_status framework function and its unit test.

### Unittests results:

```
framework/wazuh/core/tests/
===================== 662 passed, 1 xfailed, 2 xpassed, 10 warnings in 4.71s =====================


framework/wazuh/tests/
===================== 591 passed, 13 warnings in 67.11s (0:01:07) =====================

api/api/test/
===================== 225 passed, 16 warnings in 1.16s =====================

framework/wazuh/core/cluster/tests/
===================== 56 passed, 4 warnings in 0.44s =====================

framework/wazuh/core/cluster/dapi/tests/
===================== 23 passed, 9 warnings in 0.60s =====================

framework/wazuh/rbac/tests/
===================== 227 passed, 3 warnings in 178.93s (0:02:58) =====================
```


### API integration tests results:

```
test_agent_DELETE_endpoints.tavern.yaml 
	 6 passed, 8 warnings

test_agent_GET_endpoints.tavern.yaml 
	 91 passed, 93 warnings

test_security_DELETE_endpoints.tavern.yaml 
	 15 passed, 17 warnings

test_security_GET_endpoints.tavern.yaml 
	 11 passed, 13 warnings

test_security_POST_endpoints.tavern.yaml 
	 7 passed, 9 warnings

test_security_PUT_endpoints.tavern.yaml 
	 9 passed, 11 warnings
```

Regards,
Manuel.

vicferpoy reviewed (changes requested):
In the PR description you said:
> I have also updated the integration_test_api_endpoints.json file as there was a file missing.

I do not see that file on `Files changed`. Check if there was an error, please.

mcarmona99 commented:
> In the PR description you said:
> 
> > I have also updated the integration_test_api_endpoints.json file as there was a file missing.
> 
> I do not see that file on `Files changed`. Check if there was an error, please.

That line is wrong in the description. That change was supposed to be done in this PR, but at the end it was done in the following: https://github.com/wazuh/wazuh/pull/8851

Regards,
Manuel

vicferpoy reviewed (approved):
LGTM!

Selutario commented:
Hi @mcarmona99,

If possible, take a look at this code too:
https://github.com/wazuh/wazuh/blob/aab8d75e57914923a6c76399c84ebafbdba34719/framework/wazuh/core/agent.py#L47-L48

Since now the status is no longer calculated using `version`+`last_keep_alive`, maybe only `id` is necessary:
```PYTHON
        if min_select_fields is None:
            min_select_fields = {'id'}
```

Regards,
Selu.

mcarmona99 commented:
### PR update

As @selutario said in his last comment, the WazuhDBQueryAgents `min_select_fields` needs to be changed due to the new behavior. I have changed it and reviewed some old code.

I have changed the API integration tests using the select parameter to use `test_select_key_affected_items` as we were not testing them properly in some test cases.

I have also added a new test case to test the `GET /agents` endpoint sorting by status as old code to sort by `status` has been removed to.


- API integration tests results:

```
test_agent_DELETE_endpoints.tavern.yaml 
	 6 passed, 8 warnings

test_agent_GET_endpoints.tavern.yaml 
	 91 passed, 93 warnings

test_agent_POST_endpoints.tavern.yaml 
	 5 passed, 7 warnings

test_agent_PUT_endpoints.tavern.yaml 
	 10 passed, 12 warnings
```

- Unittests results:

```
framework/wazuh/core/tests/
=================== 662 passed, 1 xfailed, 2 xpassed, 9 warnings in 4.21s ===================

framework/wazuh/tests/
=================== 591 passed, 13 warnings in 59.92s ===================
```



Regards,
Manuel.

vicferpoy reviewed (dismissed):
LGTM!
